docs/source/modin/performance.rst

```diff
@@ -0,0 +1,58 @@
+Performance Recommendations
+===========================
+
+This page contains recommendations to help improve performance when using the Snowpark pandas API.
+
+Caching Intermediate Results
+----------------------------
+Snowpark pandas uses a lazy paradigm - when operations are called on a Snowpark pandas object,
+a lazy operator graph is built up and executed only when an output operation is called (e.g. printing
+the data, or persisting it to a table in Snowflake). This paradigm mirrors the Snowpark DataFrame paradigm,
+and enables larger queries to be optimized using Snowflake's SQL Query Optimizer. Certain workloads; however,
+can generate large operator graphs that include repeated, computationally expensive, subgraphs.
+Take the following code snippet as an example:
+
+.. code-block:: python
+
+    import modin.pandas as pd
+    import numpy as np
+    import snowflake.snowpark.modin.plugin
+    from snowflake.snowpark import Session
+
+    # Session.builder.create() will create a default Snowflake connection.
+    Session.builder.create()
+    df = pd.concat([pd.DataFrame([range(i, i+5)]) for i in range(0, 150, 5)])
+    print(df)
+    df = df.reset_index(drop=True)
+    print(df)
+
+The above code snippet creates a 30x5 DataFrame using concatenation of 30 smaller 1x5 DataFrames,
+prints it, resets its index, and prints it again. The concatenation step can be expensive, and is
+lazily recomputed every time the dataframe is materialized - once per print. Instead, we recommend using
```

Comment:
We should also give user a tool to check how the complexity looks like? E.g., the number of api calls stored in the current df, then they can decide whether they should cache?

No need to block this PR but we should think about this. 
cc @sfc-gh-jkew @sfc-gh-helmeleegy

Reply:
Can open up a ticket to get to that after this: https://snowflakecomputing.atlassian.net/browse/SNOW-1445508